### ISSUE TITLE
Passman hangs "unlocking" the password store

### ISSUE THREAD

**Issue by Larx** [brantje/passman#125]
I've been using current Git for quite some time now without significant problems.

All of sudden - no OC update, maybe some app updates, I cannot pinpoint that - passman hangs when I want to access the passwords. The "loading... - circle" appears, but it is not animated, and nothing more happens. Also no entry at all in the owncloud log. Happens on different (Linux-) clients and Android (all with Firefox) as well as with Chromium in Linux. Restarting server and so on did not help.

Any help appreciated to recover the passwords :-/ ...

**Comment by brantje** [comment]
Do you have a console log? (F12 in chrome)

**Comment by Larx** [comment]
One error:
'Range.detach' is now a no-op, as per DOM (http://dom.spec.whatwg.org/#dom-range-detach).

**Comment by Larx** [comment]
One error from Firefox's console (German):

Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src https://myhost.domain.eu:9999 'unsafe-eval'").

("The settings of the page have blocked the loading of a ressource on self" - difficult to translate, as the German sentence is not correct...)

**Comment by brantje** [comment]
Hmm the CSP error -_-. I've fixed that in 9a90915b8eda8703f3b0f5237c52217a7e3220ad
Can you do a git pull and try again?

**Comment by Larx** [comment]
I already have the added line in that file. I use the git version, and it worked until a few days ago. No change to passman, maybe some change to owncloud or apache (updating openSUSE Tumbleweed rolling release)?

Maybe it's because I use a self-signed certificate or a custom port?

**Comment by brantje** [comment]
Do you use an proxy orso?

**Comment by Larx** [comment]
No, nothing. Self-signed certificate and custom SSL-port. But that was no problem until this bug appeared.

**Comment by brantje** [comment]
Can you add the following
```
$csp->allowInlineScript(true);
$csp->addAllowedScriptDomain('\'self\'');
````
Below the other $csp lines?

**Comment by Larx** [comment]
Assuming you mean the file "pagecontroller.php", the addition of these lines didn't help unfortunately.

**Comment by Larx** [comment]
Some entry from the error log:

[Wed Sep 02 14:13:57.729593 2015] [access_compat:error] [pid 23360] [client 92.213.212.54:47726] AH01797: client denied by server configuration: /srv/external/apps, referer: https://mydomain.eu:8443/owncloud/index.php/apps/passman/

I get the same error all the time with webdav, caldav, and other owncloud components, so I basically ignore it since months. Everything works, or in the case of passman, used to work. The strange thing is that there is in fact in file like the one which is denied, as I use owncloud in a subfolder (/srv/external/apps should be /srv/external/owncloud/apps). I don't know WTF owncloud leaves out the subfolder for some of the requests...

I don't know if this error is related to my passman problem, however.

**Comment by sbrandexed** [comment]
@nickvergessen Do you might know what is happening and why ownCloud is going out of the sub folder?

**Comment by Larx** [comment]
I don't know if the message is related to the passman problem, however. 
Just tried out a change in config.php:

'overwrite.cli.url' => 'https://mydomain.eu:8443/owncloud',
'overwritewebroot' => '/owncloud',

Error still there...

**Comment by Larx** [comment]
Another attempt with some partial success: I disabled the asset pipeline. Now passman still hangs when starting, however the "please wait" circle with "loading..." is no longer frozen, but rotating. However, nothing more happens :(.

This is only in Firefox, however, just tried out Rekonq just for the fun of it, and here the rotating circle is still frozen.

**Comment by animalillo** [comment]
@Larx  what version of firefox are you running? Everything works fine from other browsers? What browsers have you tested?

**Comment by Larx** [comment]
Firefox, regularly updated, must be 40.x or something - and the previous versions since I filed this book. OS Linux and Android. Chromium and Rekonq with Linux - same situation.

**Comment by animalillo** [comment]
Maybe you have some plugin interfering with passman in your firefox, i use firefox from both, windows and linux systems and passman works.

**Comment by Larx** [comment]
As I have the problem with several browsers, and as AdBlock is my only FF plugin (which I deactivated, too), I don't think it's a FF problem.
